src/robot_arm.hpp

```diff
@@ -106,6 +149,42 @@ class RobotArm {
      */
     char *intToChar(int x, char *dest);
 
+    /**
+     * @brief Convert a string to a integer.
+     *
+     * This method only strings containing integers, like "12345". We cast it to a 12345 integer.
+     *
+     * @param str Input string.
+     * @param posStart
+     * @param posEnd
+     * @return int char
+     */
+    int charToInt(const char *str, const unsigned int posStart, const unsigned int posEnd) const;
+
+    /**
+     * @brief Receive Gcode string from the uArm Swift Pro using UART.
+     *
+     * We continuely poll the uArm Swift Pro for new serial data. If the read timeout is reached,
+     * we will stop the polling and return 0 (no characters received).
+     *
+     * @param response Gcode response string.
+     * @param responseSize Gcode response string size.
+     * @param readTimeout UART receiver timeout in milliseconds.
+     * @return int Amount of character read (including \0).
+     */
+    int receiveGcodeResponse(char *response, size_t responseSize, unsigned int readTimeout = 50);
+
+    /**
+     * @brief Check if the uArm Swift Pro is connected.
+     *
+     * By trying to receive the firmware version, we determine if the uArm Swift Pro is connected.
+     * If the arm is not connected, a serial receive timeout will occur.
+     *
+     * @return true Device connected.
```

Comment:
Please use only 1 @return command.
e.g. @return Boolean on whether the device is connected or not.